src/com/google/javascript/jscomp/deps/ModuleLoader.java

```diff
@@ -388,38 +417,66 @@ public static boolean isPathIdentifier(String name) {
     return name.contains(MODULE_SLASH);
   }
 
+  /** Removes the leading slash from absolute identifiers */
+  public static String removeAbsoluteIdentifierIndicator(String name) {
+    if (isAbsoluteIdentifier(name)) {
+      return name.substring(1);
+    }
+    return name;
+  }
+
+  /** Converts paths which don't start with either a '.' or '/' to absolute paths */
+  public static String toAbsoluteIdentifier(String name) {
+    if (name.length() == 0 || isAbsoluteIdentifier(name) || isRelativeIdentifier(name)) {
+      return name;
```

Comment:
> How about we detect ambiguous paths ASAP and either report an error or explicitly force them to be relative or absolute at that point?

I think ambiguous paths (as you labeled them) are used widely inside google. For instance, this path is ambiguous:

```js
import {foo} from 'path/to/bar'
```

I can force them to be absolute if you are willing to try landing it ...